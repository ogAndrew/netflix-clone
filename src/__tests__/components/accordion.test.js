import React from 'react';
import { render, fireEvent, queryByTestId } from '@testing-library/react';
import faqsData from '../../fixtures/faqs';
import { Accordion } from '../../components';

describe('<Accordion />', () => {
  it('renders the <Accordion /> with populated data', () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Frame>
          {faqsData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Frame>
      </Accordion>
    );

    expect(getByText('Frequently Asked Questions')).toBeTruthy();
    expect(getByText('What is Netflix?')).toBeTruthy();
    expect(getByText('How much does Netflix cost?')).toBeTruthy();
    expect(getByText('Where can I watch?')).toBeTruthy();
    expect(getByText('How do I cancel?')).toBeTruthy();
    expect(getByText('What can I watch on Netflix?')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('opens and closes the <Accordion /> component', () => {
    const { container, queryByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Frame>
          {faqsData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Frame>
      </Accordion>
    );

    const whatIsNetflixBodyText =
      "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

    expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
    fireEvent.click(queryByText('What is Netflix?'));
    expect(queryByText(whatIsNetflixBodyText)).toBeTruthy();

    fireEvent.click(queryByText('What is Netflix?'));
    expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();

    const howMuchDoesNetfliCostBodyText =
      'Watch Netflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from £5.99 a month. No extra costs or contracts.';

    expect(queryByText(howMuchDoesNetfliCostBodyText)).toBeFalsy();
    fireEvent.click(queryByText('How much does Netflix cost?'));
    expect(queryByText(howMuchDoesNetfliCostBodyText)).toBeTruthy();

    fireEvent.click(queryByText('How much does Netflix cost?'));
    expect(queryByText(howMuchDoesNetfliCostBodyText)).toBeFalsy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
