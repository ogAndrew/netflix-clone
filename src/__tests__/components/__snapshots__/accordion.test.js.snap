// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-sc-1wnyt6p-0 cNKPUD"
>
  <div
    class="accordion__Inner-sc-1wnyt6p-2 ggIgBn"
  >
    <h1
      class="accordion__Title-sc-1wnyt6p-3 ddnMFY"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Frame-sc-1wnyt6p-1 lbCZoO"
    >
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sc-1wnyt6p-0 cNKPUD"
>
  <div
    class="accordion__Inner-sc-1wnyt6p-2 ggIgBn"
  >
    <h1
      class="accordion__Title-sc-1wnyt6p-3 ddnMFY"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Frame-sc-1wnyt6p-1 lbCZoO"
    >
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1wnyt6p-4 iMllok"
      >
        <div
          class="accordion__Header-sc-1wnyt6p-5 fHKeEt"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
